veda delivery worker role

muhammad-ammar opened the pull request:
EDUCATOR-1665

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

muhammad-ammar commented:
@feanil This is ready for review. Sandbox build is [successful](https://tools-edx-jenkins.edx.org/job/Sandboxes/job/CreateSandbox/6740/console).

feanil commented on playbooks/roles/veda_delivery_worker/tasks/main.yml:
```diff
@@ -1 +1,109 @@
+    owner: "{{ veda_delivery_worker_service_name }}"
+    group: "{{ common_web_group }}"
+  with_items:
+    - "VEDA_WORKING"
```
What's the point of this directory?  If it's for temp files and such,  It should probably go under the `/edx/var` directory instead of the app directory.

muhammad-ammar replied:
@feanil edx-video-pipeline creates/expects this directory in the app directory. We will improve this once relevant code in edx-video-pipeline is refactored.

feanil replied:
Sounds good.  Can you add a todo with the ticket that will fix this in the future?

muhammad-ammar replied:
@feanil Added [TODO](https://github.com/edx/configuration/pull/4209/files#diff-184879b2f04049b739fdaffb16d2cab3R24)

feanil commented on playbooks/roles/veda_delivery_worker/templates/edx/app/veda_delivery_worker/veda_delivery_worker.sh.j2:
```diff
@@ -0,0 +1,18 @@
+
+{% set veda_delivery_worker_venv_bin = veda_delivery_worker_home + '/venvs/' + veda_delivery_worker_service_name + '/bin' %}
+{% if COMMON_ENABLE_NEWRELIC_APP %}
+{% set executable = veda_delivery_worker_venv_bin + '/newrelic-admin run-program ' + veda_delivery_worker_venv_bin + '/python' %}
```
This is the app I was thinking about when I mentioned in the other PR that we aren't ready to deal with New Relic for some of the veda pieces.  I think you should remove all the newrelic related config from this PR for now.

feanil commented on playbooks/roles/video_pipeline_base/meta/main.yml:
```diff
@@ -2,3 +2,6 @@
   - common
+
+# Allow this role to be duplicated in dependencies.
+allow_duplicates: yes
```
Why do you need this for this role?  I wouldn't think two runs of this role should happen on the same machine.

muhammad-ammar replied:
@feanil In the sandbox video pipeline workers and roles will be using this base role and it will be executed multiple times.

muhammad-ammar commented:
@feanil Addressed your feedback and built the [sandbox](https://tools-edx-jenkins.edx.org/job/Sandboxes/job/CreateSandbox/6989/console) as per step 2.
